libs/exec/execv_windows.go

```diff
@@ -0,0 +1,95 @@
+//go:build windows
+
+package exec
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"os/exec"
+	"strings"
+	"sync"
+)
+
+// Note: Windows does not support an execv syscall that replaces the current process.
+// Thus we emulate that by launching a child process and streaming the output and returning
+// the exit code.
+// ref: https://learn.microsoft.com/en-us/cpp/c-runtime-library/reference/execv-wexecv?view=msvc-170
+func execv(opts ExecvOptions) error {
+	path, err := exec.LookPath(opts.Args[0])
+	if err != nil {
+		return fmt.Errorf("looking up %q failed: %w", opts.Args[0], err)
+	}
+
+	cmd := exec.Command(path, opts.Args[1:]...)
+
+	cmd.Dir = opts.Dir
+	cmd.Env = opts.Env
+
+	stdout, err := cmd.StdoutPipe()
+	if err != nil {
+		return fmt.Errorf("creating stdout pipe failed: %w", err)
+	}
+
+	stderr, err := cmd.StderrPipe()
+	if err != nil {
+		return fmt.Errorf("creating stderr pipe failed: %w", err)
+	}
+
+	// Start the child command.
+	err = cmd.Start()
+	if err != nil {
+		return fmt.Errorf(" %s failed: %w", strings.Join(opts.Args, " "), err)
+	}
+
+	var wg sync.WaitGroup
+	wg.Add(2)
+
+	var stdoutErr error
+	go func() {
+		defer wg.Done()
+
+		scanner := bufio.NewScanner(stdout)
```

Comment:
Do we need to read full lines before we process it?

can you add a test that uses something like printf to output a line without newline at the end?

/tmp % echo "hello"
hello
/tmp % printf "hello"
hello%

Reply:
> Do we need to read full lines before we process it?

Technically not but there's not much foreseeable benefit in streaming character by character instead of line by line. Can be an incremental followup if needed.

Added a test case in bundle/run/inline-script/basic

Reply:
It's not character by character, it's by whatever chunks the program sends. 

It's not unusual for program to output partial lines, e.g.
Waiting for remote.... <5 min wait> prints newline.

Reply:
(This discussion won't be necessary if you make it inherit FDs and don't read that at all).

Reply:
Modified the function to stream chunks as they come in.